### ISSUE TITLE
Data Collector: Read CollectionType

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/libSplash#189]
The `read` and `readAttribute` calls in the [DataCollector](https://computationalradiationphysics.github.io/libSplash/classsplash_1_1_data_collector.html) interface lack, in comparison to their write counter-parts, the possibility to read the `CollectionType` again.

It would be very helpful to take the maximum arguments in one of the two read calls and to add an additional call to read everything *but* the data itself. I am thinking about a virtual interface such as:
```c++
readMeta(int32_t id,
         CollectionType &type,
         const char* name,
         const Dimensions dstBuffer,
         const Dimensions dstOffset,
         Dimensions &sizeRead) throw (DCException);
```
for `DataCollector`.

*Note:* Currently all of the information of such an interface, but not the `type`, can be get from the `read()` interface with a `NULL` data pointer. An alternative would be to extend that interface, but this would probably require a `CollectionType*` argument for the type which again might (?) need an allocation *inside* the member which is ugly (since it is not required for a read if the type is known and won't help because it's to late to allocate data for it; it also tends to create memory leaks in the user code).

*Alternative:* It would also be quite nice if either `CollectionType` or/and `DCDataType` and the buffer `Dimensions` could be filled in the `DCEntry` itself, this would also work and no extension of the API is necessary (but I found it a bit more cumbersome when trying to implement that approach).

*P.S.:* `DCDataType` and `DCDataSet` lack a doxygen documentation in the Doxygen `Class List`.

**Comment by ax3l** [comment]
Draft commit for further development: https://github.com/ax3l/libSplash/commit/95b42572ee485141508fbf6e35d98320f0e21d66

**Comment by ax3l** [comment]
closed with #203
